File: src/문자열/q_1283.java
package 문자열;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class q_1283 {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        boolean[] used = new boolean[26];
        StringBuilder sb = new StringBuilder();

        for (int t = 0; t < n; t++) {
            String line = br.readLine();
            int idx = -1;

            StringTokenizer st = new StringTokenizer(line);
            int pos = 0;

            while (st.hasMoreTokens()) {
                String word = st.nextToken();
                while (pos < line.length() && line.charAt(pos) == ' ') {
                    pos++;
                }

                char c = line.charAt(pos);
                int alpha = Character.toLowerCase(c) - 'a';

                if (!used[alpha]) {
                    used[alpha] = true;
                    idx = pos;
                    break;
                }

                pos += word.length();
            }

            if (idx == -1) {
                for (int i = 0; i < line.length(); i++) {
                    char c = line.charAt(i);

                    if (Character.isAlphabetic(c)) {
                        int alpha = Character.toLowerCase(c) - 'a';
                        if (!used[alpha]) {
                            used[alpha] = true;
                            idx = i;
                            break;
                        }
                    }
                }
            }

            if (idx == -1) {
                sb.append(line).append("\n");
            } else {
                sb.append(line, 0, idx)
                        .append('[')
                        .append(line.charAt(idx))
                        .append(']')
                        .append(line.substring(idx + 1))
                        .append("\n");
            }

        }

        System.out.println(sb);


    }
}

File: kotlin/문자열/q_1283.kt
package 문자열

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.StringTokenizer

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val n = br.readLine().toInt()

    val used = BooleanArray(26)
    val sb = StringBuilder()

    repeat(n) {
        val line = br.readLine()
        var idx = -1

        val st = StringTokenizer(line)
        var pos = 0
        while (st.hasMoreTokens()) {
            val word = st.nextToken()

            while (pos < line.length && line[pos] == ' ') pos++

            val alpha = line[pos].lowercaseChar() - 'a'
            if (!used[alpha]) {
                used[alpha] = true
                idx = pos
                break
            }

            pos += word.length
        }

        if (idx == -1) {
            for (i in line.indices) {
                if (line[i].isLetter()) {
                    val alpha = line[i].lowercaseChar() - 'a'
                    if (!used[alpha]) {
                        used[alpha] = true
                        idx = i
                        break
                    }
                }
            }
        }

        if (idx == -1) {
            sb.append(line).append("\n")
        } else {
            sb.append(line.substring(0, idx))
                .append('[')
                .append(line[idx])
                .append(']')
                .append(line.substring(idx + 1))
                .append("\n")
        }


    }
    print(sb)
}
